cityofaustin/atd-moped
# Add Knack Signal ID to the Moped Component Payload & ETL Script

## Associated issues

This PR intends to close https://github.com/cityofaustin/atd-data-tech/issues/7925.

Please see [child issue #8213](https://github.com/cityofaustin/atd-data-tech/issues/8213) for additional work which has been requested related to this.

## Testing

https://deploy-preview-522--atd-moped-main.netlify.app/moped/

I find this easier to test locally as you may have to manipulate some records to facilitate testing, but it should work on the other deployments. 

* Visit knack builder's development database for moped ("TEST | Austin Transportation Data Tracker | MOPED DEV - 12 OCT 2021") and clear out any records that have a Moped Project ID. 
* Similarly, make sure you have no records in the Moped Database which indicate they have been sync'd to Knack by looking for projects with a `knack_project_id`
* Add some signals to a project that has not been sync'd
* Sync them using the UI element on the project summary page.
* Observe the record created in the Knack app

## :warning: Please see [a warning to testers](https://github.com/cityofaustin/atd-moped/pull/522#issuecomment-1033232585) related to signal IDs & Knack.

#### Ship list
